components/micro-gateway-core/src/main/ballerina/src/gateway/utils/analytics_util.bal

```diff
@@ -35,7 +36,13 @@ function populateThrottleAnalyticsDTO(http:FilterContext context) returns (Throt
     APIConfiguration? apiConfiguration = apiConfigAnnotationMap[context.getServiceName()];
     if (apiConfiguration is APIConfiguration) {
         eventDto.apiVersion = apiConfiguration.apiVersion;
-    }
+        if (!stringutils:equalsIgnoreCase("", <string>apiConfiguration.publisher)) {
```

Comment:
duplicated code in populateThrottleAnalyticsDTO and populateFaultAnalyticsDTO. I think you can use a common method for them.